src/OctopusPuppet.OctopusProvider/OctopusDeploymentPlanner.cs

```diff
@@ -126,11 +126,7 @@ private Component GetComponentForBranch(DashboardResource dashboard, string envi
             
             var projectVariables = _repository.VariableSets.Get(releaseResource.ProjectVariableSetSnapshotId);
 
-            var componentDependancies = projectVariables.Variables
-                .Where(x => x.Name == ComponentDependancies)
-                .SelectMany(x => JsonConvert.DeserializeObject<string[]>(x.Value))
-                .Where(x => componentFilter == null || componentFilter.Match(x))
-                .ToList();
+            var componentDependancies = GetComponentDependancies(componentFilter, projectVariables, releaseResource.Id);
```

Comment:
componentDependencies is actually a list of variables matching componentDependencies. Need renaming here and below. Suggestion 'variablesWithComponentDependencies' or 'componentDependencyVariables'